### PR TITLE
Add Tests

### PR DESCRIPTION
**Pull request by kd7lxl**
I'm planning to add a feature to tugger, and before I get started I wanted a good regression suite. This PR adds a number of test cases, bringing test coverage from 0% to around 70%. I found a couple bugs in the process and included fixes for those.

### CODE REVIEW

**Comment by jainishshah17** [comment]
@kd7lxl Can you please bump the application and chart version as you have modified the functionality for https://github.com/jainishshah17/tugger/issues/17. Thanks

**Comment by kd7lxl** [comment]
> @kd7lxl Can you please bump the application and chart version as you have modified the functionality for #17. Thanks

No problem. Let me know if this looks alright.

File: chart/tugger/values.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ serviceAccount:
 
 image:
   repository: jainishshah17/tugger
-  tag: 0.0.6
+  tag:
```
Comment (kd7lxl) [inline chart/tugger/values.yaml:22]: Since the deployment defaults the image to the appVersion, I was able to remove this reference.

**Comment by jainishshah17** [comment]
LGTM

**Pull request merged**